src/plugins/intel_cpu/src/nodes/mkldnn_tensoriterator_node.cpp

```diff
@@ -567,33 +604,30 @@ void MKLDNNTensorIteratorNode::prepareDynamicBackEdges() {
     back_mappers.clear();
     for (auto map_rule : backEdges) {
         auto from_mem = output_mem[map_rule.from];
-        auto to_mem = input_mem[map_rule.to];
+        auto to_mems = input_mems[map_rule.to];
 
-        // need to reshape body inputs by output if body has internal dynamism
-        const auto &currDesc = to_mem->getDesc();
-        if (currDesc.getShape().isDynamic() || currDesc.getShape().getStaticDims() != from_mem->getStaticDims()) {
-            const auto& memDesc = from_mem->getDesc();
-            to_mem->redefineDesc(memDesc);
-        }
+        const auto& desc = from_mem->getDesc();
+        redefineToMemories(to_mems, desc.clone());
 
-        back_mappers.emplace_back(std::make_shared<BackEdgePortHelper>(from_mem, to_mem, eng));
+        // first memory is enough to get common memory ptr
+        back_mappers.emplace_back(std::make_shared<BackEdgePortHelper>(from_mem, to_mems.front(), eng));
     }
 }
 
 void MKLDNNTensorIteratorNode::prepareDynamicBuffers() {
     for (auto map_rule : outputPortMap) {
         if (map_rule.axis != -1) {
-            auto &to_mem = getChildEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
+            auto to_mems = getToMemories(this, map_rule.from);
             auto &from_mem = output_mem[map_rule.to];
-            buffers.emplace_back(std::make_shared<DynamicBuffer>(from_mem, to_mem, map_rule));
+            buffers.emplace_back(std::make_shared<DynamicBuffer>(from_mem, to_mems, map_rule));
         }
     }
 }
 
 void MKLDNNTensorIteratorNode::prepareLoopBodyCurrentIteration() {
     const auto &eng = getEngine();
     for (auto idx : loopBodyCurrentIterationIdx) {
-        auto to_mem = input_mem[idx];
+        auto to_mem = input_mems[idx].front();  // first memory is enough to get common memory ptr
```

Comment:
If the first memory pointer is usually enough, m.b. we do not need to use the 2d memory ptrs array, and can return to the previous solution where we had 1d array of mem pointers with size of the ports number?